stmhal/dac.c

```diff
@@ -171,6 +171,14 @@ STATIC mp_obj_t pyb_dac_init_helper(pyb_dac_obj_t *self, mp_uint_t n_args, const
     #endif
 
     // stop anything already going on
+    __DMA1_CLK_ENABLE();
+    DMA_HandleTypeDef DMA_Handle;
```

Comment:
What happens if dma_idle_handler() disables the DMA between here and the call to HAL_DMA_DeInit?